app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,73 @@
+class MoviesController < ApplicationController
+  before_action :is_admin?
+
+
+  def index
+    if user_signed_in?
+      @movies = Movie.all
+      render 'index'
+    else
+      redirect_to new_user_session_path
+    end
+  end
+
+  def new
+      @movie =Movie.new
+  end
+
+  def create
+    @movie = Movie.new(movie_param)
+    if @movie.save
+      redirect_to movies_path
+    else
+      redirect_to new_movie_path
+    end
+  end
+
+  def edit
+    @movie=Movie.find(params[:id])
+  end
+
+  def update
+    @movie = Movie.find(params[:id])
+    if @movie.update(movie_param)
+      redirect_to movies_path
+    else
+      render 'edit'
+    end
+  end
+
+
+  def destroy
+    @movie = Movie.find(params[:id])
+    if @movie.destroy
```

Comment:
If `@movie.destroy` fails, it would throw a render error, so either this conditional is redundant or we should add an else condition to it with a flash message